nano/node/block_broadcast_strategy.cpp

```diff
@@ -0,0 +1,51 @@
+#include <nano/node/block_arrival.hpp>
+#include <nano/node/block_broadcast_strategy.hpp>
+#include <nano/node/blockprocessor.hpp>
+#include <nano/node/network.hpp>
+
+nano::block_broadcast_strategy::block_broadcast_strategy (nano::network & network, nano::block_arrival & block_arrival) :
+	network{ network },
+	block_arrival{ block_arrival }
+{
+}
+
+void nano::block_broadcast_strategy::connect (nano::block_processor & block_processor, bool enabled)
+{
+	if (!enabled)
+	{
+		return;
+	}
+	block_processor.processed.add ([this] (auto const & result, auto const & block) {
+		switch (result.code)
+		{
+			case nano::process_result::progress:
+				observe (block);
+				break;
+			default:
+				break;
+		}
+	});
+}
+
+void nano::block_broadcast_strategy::observe (std::shared_ptr<nano::block> block)
+{
+	nano::lock_guard<nano::mutex> lock{ mutex };
+	auto existing = local.find (block);
+	if (existing != local.end ())
```

Comment:
is the lock necessary after `local.find (block)`?

Reply:
Modified it to release after checking existence.